src/OrbitGl/TrackHeader.cpp

```diff
@@ -0,0 +1,151 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "TrackHeader.h"
+
+#include "AccessibleCaptureViewElement.h"
+#include "GlCanvas.h"
+#include "Track.h"
+#include "TrackRenderHelper.h"
+
+namespace orbit_gl {
+
+TrackHeader::TrackHeader(CaptureViewElement* parent, const Viewport* viewport,
+                         const TimeGraphLayout* layout)
+    : CaptureViewElement(parent, viewport, layout) {
+  ORBIT_CHECK(dynamic_cast<Track*>(parent) != nullptr);
+
+  collapse_toggle_ = std::make_shared<TriangleToggle>(
+      this, viewport, layout, [this](bool /*is_collapsed*/) { RequestUpdate(); });
+}
+
+void TrackHeader::OnPick(int x, int y) {
+  CaptureViewElement::OnPick(x, y);
+  GetParent()->OnPick(x, y);
+}
+
+std::unique_ptr<orbit_accessibility::AccessibleInterface> TrackHeader::CreateAccessibleInterface() {
+  return std::make_unique<orbit_gl::AccessibleCaptureViewElement>(
+      this, dynamic_cast<Track*>(GetParent())->GetName() + "_tab",
```

Comment:
Following the comment above, as this is the first usage of the parent: I think it is totally fine to restrict the generic parent/child relationship. But I really think we should manifest such a restriction on the type level.

The dynamic casts here are not only a potential source of bugs, but more importantly hinder readability/understandability of the code.

Reply:
I am not really sure anymore if this was discussed. I think I had the need for this first, when I was working on the Gpu tracks. For the argument above (making the restriction/assumption visible on the type level), this is the only legit option to me, and I am fine with the internal private shadow of the parent_ member.